src/app/events/[slug]/components/ScheduleTabs.tsx

```diff
@@ -0,0 +1,99 @@
+'use client'
+
+import { useEffect, useRef } from 'react'
+
+import { Tab, TabGroup, TabList, TabPanel, TabPanels } from '@headlessui/react'
+import theme from 'tailwindcss/defaultTheme'
+
+import type { Event } from '@/types/eventType'
+
+import { BasicCard } from '@/components/BasicCard'
+import { Heading } from '@/components/Heading'
+import { TextLink } from '@/components/TextLink'
+
+import { formatDate, formatTime } from '../utils/dateUtils'
+
+type ScheduleTabsProps = {
+  schedule: Event['schedule']
+}
+
+const { screens } = theme
+
+export function ScheduleTabs({ schedule }: ScheduleTabsProps) {
+  const tabGroupRef = useRef<HTMLDivElement>(null)
+  const hasMounted = useRef(false)
+
+  const validDays = schedule!.days.filter((day) => day.events.length > 0)
+
+  useEffect(() => {
+    hasMounted.current = true
+  }, [])
```

Comment:
Nice little optimisation - we can import `useIsMounted` from [usehooks-ts](https://usehooks-ts.com/react-hook/use-is-mounted), which is already part of the project :)